File: SakhaTyla.Web.Ng/src/app/book-label-core/book-label.model.ts
import { OrderDirection } from '../core/models/order-direction.model';
import { PageSettings } from '../core/page.model';
import { BookLabelFilter } from './book-label-filter.model';

export class BookLabel {
  id: number;
  creationDate: Date;
  modificationDate: Date;
  creationUserId: number;
  modificationUserId: number;
  bookId: number;
  name: string;
  pageId: number;

  constructor(bookId: number) {
    this.bookId = bookId;
  }
}

export class BookLabelListState {
  pageSize = PageSettings.pageSize;
  pageIndex = 0;
  filter = new BookLabelFilter();
  orderBy?: string;
  orderDirection?: OrderDirection;
}

File: SakhaTyla.Core/Requests/Public/BookLabels/Models/BookLabelModel.cs
namespace SakhaTyla.Core.Requests.Public.BookLabels.Models
{
    public class BookLabelModel
    {
        public BookLabelModel(string name)
        {
            Name = name;
        }

        public int Id { get; set; }

        public int BookId { get; set; }

        public string Name { get; set; }

        public int PageId { get; set; }
    }
}
